packages/protocol/contracts/common/OwnerUUPSUpgradable.sol

```diff
@@ -44,7 +44,7 @@ abstract contract OwnerUUPSUpgradable is UUPSUpgradeable, OwnableUpgradeable {
         if (_loadReentryLock() == _TRUE) revert REENTRANT_CALL();
         _storeReentryLock(_TRUE);
         _;
-        _storeReentryLock(_FALSE);
```

Comment:
I wonder if this `_storeReentryLock(_FALSE)` call is even necessary. A transient storage is cleared automatically at the end of the transition, no?

Reply:
Transient stroage cleared when transaction ends (for all the contracts). But contract A might call B's function (which is `nonReentrant`) multiple times. So if we don't clear, the second call would fail.

<img width="731" alt="kép" src="https://github.com/taikoxyz/taiko-mono/assets/51912515/d346b20c-5c6c-4968-ab31-35fdce2a641d">